packages/core/typescript/TypeScriptFileUpdate.ts

```diff
@@ -313,7 +316,11 @@ export class TypeScriptFileUpdate {
 		let className: string;
 		const fileSource = TsUtils.getFileSource(filePath);
 		const relativePath: string = Util.relativePath(this.targetPath, filePath, true, true);
-		className = TsUtils.getClassName(fileSource.getChildren());
+		if (providedClassName != "") {
+            className = providedClassName;
+        } else {
+            className = TsUtils.getClassName(fileSource.getChildren());
```

Comment:
Probably a better idea is to do `className = providedClassName ||  TsUtils.getClassName(fileSource.getChildren());` and ditch the rest of the `if` statement. Since that would make sure any `falsy` value is accounted for. With the current approach, if I pass in values such as `undefined` or `null`, that would cause the runtime to enter in the first block of the `if` statement since both are technically not empty strings. Which would result in an empty `className` and potential errors afterwards.